storage/s3/src/main/java/io/aiven/kafka/tieredstorage/storage/s3/S3Storage.java

```diff
@@ -62,11 +64,30 @@ public void configure(final Map<String, ?> configs) {
 
     @Override
     public long upload(final InputStream inputStream, final ObjectKey key) throws StorageBackendException {
+        try {
+            final int availableSize = inputStream.available();
+            if (availableSize <= partSize) {
```

Comment:
`InputStream#available` doesn't guarantee to return the total size. This is one of the reasons I was suggesting the S3UploadOutputStream approach as it is already reading the stream and the final size is available there.

Reply:
I was also worry about it and tested with deploy. it passed all test. Anyway. it is risk if we don't want to check the caller's code. Let me consider your proposal and make one code sample without test.

Reply:
@jeqo  done with all code/test. help to review. thanks